### PR TITLE
[Master] Make port_macs as the optional argument in guest_register API

### PR DESCRIPTION
**Pull request by SeanHQF**
Sorry that the previous PR https://github.com/openmainframeproject/python-zvm-sdk/pull/76
has bug on handling port_macs as the optional argument in guest_register API,
this PR makes it as the optional argument and also provides the fix for the key error
if mac key doesn't exist in nics_info or port_macs argument.

All local check passed with 'tox -v' as attached and local function test also passed and here is the test code:
#!/usr/bin/env python
import sys
from zvmconnector import connector
import six
from nova import exception

conn = connector.ZVMConnector('9.152.85.144', port=8080, token_path='/etc/zvmsdk/token.dat')
results = conn.send_request('guest_list')

if results['overallRC'] != 0:
    print("zVM Cloud Connector request failed. Results: %(results)s",
                {'results': six.text_type(results)})
    raise exception.ZVMConnectorError(results=results)

print "Success! Result1: %s" % six.text_type(results)
print "============================================="
print "============================================="

#results = conn.send_request('guest_register', 'fba', 'rhel7', '1')
#results = conn.send_request('guest_register', 'fba', 'rhel7', '1', 'portttt')
results = conn.send_request('guest_register', 'fba', 'rhel7', '1', {'mac1': 'port1'})
if results['overallRC'] != 0:
    print("zVM Cloud Connector request failed. Results: %(results)s",
                {'results': six.text_type(results)})
    raise exception.ZVMConnectorError(results=results)

print "Success! Result2: %s" % six.text_type(results)
print "============================================="
results = conn.send_request('guest_deregister', 'fba')
if results['overallRC'] != 0:
    print("zVM Cloud Connector request failed. Results: %(results)s",
                {'results': six.text_type(results)})
    raise exception.ZVMConnectorError(results=results)

print "Success! Result3: %s" % six.text_type(results)
[master-fix.log](https://github.com/openmainframeproject/python-zvm-sdk/files/4249996/master-fix.log)

### CODE REVIEW

File: zvmsdk/tests/unit/sdkwsgi/handlers/test_guest.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def test_guest_register(self, mock_action,
         self.req.body = '{"action": "register_vm",\
                           "meta": "rhel7",\
                           "net_set": "1",\
-                          "port": "5abc7819-abec-4deb-9115-2af5da249155"}'
+                          "port_macs": "5abc7819-abec-4deb-9115-2af5da249155"}'
```
Comment (jichenjc) [inline zvmsdk/tests/unit/sdkwsgi/handlers/test_guest.py]: do we have a test case for not providing port_macs in unit test?

File: zvmconnector/restclient.py [hunk 2]
```diff
@@ -241,8 +241,9 @@ def req_guest_register(start_index, *args, **kwargs):
     url = '/guests/%s/action'
     body = {'action': 'register_vm',
             'meta': args[start_index],
-            'net_set': args[start_index + 1],
-            'port_macs': args[start_index + 2]}
+            'net_set': args[start_index + 1]}
```
Comment (jichenjc) [inline zvmconnector/restclient.py]: do you want to update doc?

File: zvmsdk/tests/unit/sdkwsgi/handlers/test_guest.py (hunk 1 above)
Comment (SeanHQF) [reply zvmsdk/tests/unit/sdkwsgi/handlers/test_guest.py]: No, I just added a test case of invalid port_macs in ut, is that needed? I can add one case. Function test for no port_macs has passed. thanks

Comment (jichenjc) [reply zvmsdk/tests/unit/sdkwsgi/handlers/test_guest.py]: when you say functional test , you mean manual test ? 
what can we guarantee that later on? please add ut to make sure your logic works later as well

File: zvmconnector/restclient.py (hunk 2 above)
Comment (SeanHQF) [reply zvmconnector/restclient.py]: Yes, I already set the argument in the doc as optional:
guest_register_port_macs:
  description: |
    The dict of virtual interface port id and mac id
  in: body
  required: false
  type: dict

File: zvmsdk/tests/unit/sdkwsgi/handlers/test_guest.py (hunk 1 above)
Comment (SeanHQF) [reply zvmsdk/tests/unit/sdkwsgi/handlers/test_guest.py]: Yes, manual test, I will add one ut case for that.

**Comment by SeanHQF** [comment]
> some comments above
Thanks and all addressed with code updated.

**Comment by SeanHQF** [comment]
[master-all.log](https://github.com/openmainframeproject/python-zvm-sdk/files/4253413/master-all.log)

**Comment by jichenjc** [comment]
good to go, just please fix merge conflict

**Comment by SeanHQF** [comment]
@jichenjc @bjhuangr can you please help to review and merge it? thanks

**Pull request merged**